src/glue/jobs/json_to_parquet.py

```diff
@@ -177,49 +179,52 @@ def drop_table_duplicates(
 
     Args:
         table (DynamicFrame): The table from which to drop duplicates
-        table_name (str): The name of the Glue table. Helps determine
+        data_type (str): The data type. Helps determine
             the table index in conjunction with INDEX_FIELD_MAP.
-        glue_context (GlueContext): The glue context
         record_counts (dict[str,list]): A dict mapping data types to a list
             of counts, each of which corresponds to an `event` type.
         logger_context (dict): A dictionary containing contextual information
             to include with every log.
 
     Returns:
-        awsglue.DynamicFrame: The `table` DynamicFrame after duplicates have been dropped.
+        pyspark.sql.dataframe.DataFrame: A Spark dataframe with duplicates removed.
     """
-    table_name_components = table_name.split("_")
-    table_data_type = table_name_components[1]
+    window_unordered = Window.partitionBy(INDEX_FIELD_MAP[data_type])
     spark_df = table.toDF()
     if "InsertedDate" in spark_df.columns:
-        sorted_spark_df = spark_df.sort(
-            [spark_df.InsertedDate.desc(), spark_df.export_end_date.desc()]
+        window_ordered = window_unordered.orderBy(
+                col("InsertedDate").desc(),
+                col("export_end_date").desc()
         )
     else:
-        sorted_spark_df = spark_df.sort(spark_df.export_end_date.desc())
-    table = DynamicFrame.fromDF(
-        dataframe=(
-            sorted_spark_df.dropDuplicates(subset=INDEX_FIELD_MAP[table_data_type])
-        ),
-        glue_ctx=glue_context,
-        name=table_name,
+        window_ordered = window_unordered.orderBy(
+                col("export_end_date").desc()
+        )
+    table_no_duplicates = (
```

Comment:
Fun fact, this is exactly what I do to create de-duplicated tables within the synapse data warehouse: https://github.com/Sage-Bionetworks/snowflake/blob/b4755ad226abe29d30308708539c99d632dd2337/synapse_data_warehouse/synapse/tables/V1.11.0__initial_creation_of_latest_tables.sql#L3-L17

Reply:
I think this way may be more common in databases, versus sorting and dropping as you would an in-memory dataframe. Perhaps it has something to do with the distributed (across partitions) nature of the data.